src/Symfony/Bridge/Doctrine/Security/User/EntityUserProvider.php

```diff
@@ -29,6 +29,9 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  * @author Johannes M. Schmitt <schmittjoh@gmail.com>
+ *
+ * @template-implements UserProviderInterface<UserInterface>
+ * @template-implements PasswordUpgraderInterface<PasswordAuthenticatedUserInterface>
  */
 class EntityUserProvider implements UserProviderInterface, PasswordUpgraderInterface
```

Comment:
This class can be generic too. `EntityUserProvider` is configured for an entity class and will only load instances of that class.

Reply:
Is 
```
 * @template TUser of UserInterface&PasswordAuthenticatedUserInterface
 *
 * @template-implements UserProviderInterface<TUser>
 * @template-implements PasswordUpgraderInterface<TUser>
 ```
 better ?

Reply:
Yes. 🙂